### DISCUSSION TITLE
persistent indexeddb

### DISCUSSION THREAD

**Discussion by ws-rush** [tauri-apps/tauri#6360] [Q&A] [answered]
as we know browsers clear indexeddb of PWA if user clear data, or even app doesnt used for a while or maybe decide remove it for speed, my question does indexeddb srorage in tauri is dependable? 

EDIT: from reading electron doesn't remove data with leveldb adapter, I need data to be persistent coz I will make app for financial data and I don't want depend on sql plugin to make it portable

**Comment by FabianLars** [comment, marked as answer]
It should be persistent, yes. At least as long as the user doesn't delete the app data on their disk themselves of course 😅

**Reply by ws-rush** [reply]
I discoverer there is a persistence API in modern browsers, with them we can check if data will stay or not

**Comment by JungleHH** [comment]
What if user uninstall the old version app and install the new version？The indexeddb always there?

**Reply by charlesschaefer** [reply]
On Windows, the user has the option to clear all the data or not. If they choose not, the data is persisted even between installations.

I've tested a lot in Linux and Android, and they work as expected when the user updates the app. On Android, though, if the user uninstalls the app first, the data will be removed. That doesn't happen on Linux and, on Windows, as I said, there's a flag to the user chose to remove or not.

**Reply by JungleHH** [reply]
@charlesschaefer 
Great! Thank you!
